shamil_latypov/logger.cpp

```diff
@@ -0,0 +1,34 @@
+#include "logger.hpp"
+#include <fstream>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+
+namespace uni_cource_cpp {
+
+void Logger::set_output_file_path(const std::optional<std::string>& file_path) {
+  if (file_path == std::nullopt) {
+    if (output_stream_.has_value()) {
+      output_stream_.value().close();
+    }
+  }
+
+  output_stream_.value() = std::ofstream(file_path.value());
+  if (!output_stream_.has_value()) {
```

Comment:
Что ты тут хочешь проверить?
```suggestion
  if (!output_stream_->is_open())
```